cmd/dashboard-redirector/main.go

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright (c) 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"strings"
+
+	crd "github.com/openshift/ansible-service-broker/pkg/dao/crd"
+
+	"github.com/sirupsen/logrus"
+)
+
+var options struct {
+	BrokerNamespace string
+	Port            int
+}
+
+func init() {
+	flag.IntVar(&options.Port, "port", 1337, "port that the dashboard-redirector should listen on")
+	flag.StringVar(&options.BrokerNamespace, "namespace", "ansible-service-broker", "namespace that the broker resides in")
+	flag.Parse()
+}
+
+var crdDao *crd.Dao
+
+func main() {
+	var err error
+
+	crdDao, err = crd.NewDao(options.BrokerNamespace)
+	if err != nil {
+		panic(fmt.Sprintf("Unable to create crd client - %v", err))
+	}
+
+	logrus.Info("Trying to load batch specs as a sanity check...")
+	specs, err := crdDao.BatchGetSpecs("")
+	if err != nil {
+		panic(fmt.Sprintf("Sanity check failed! -> %v", err))
+	} else {
+		logrus.Info("Sanity check passed! Loaded specs: %v", specs)
+	}
+
+	http.HandleFunc("/", redirect)
+	portStr := fmt.Sprintf(":%d", options.Port)
+
+	logrus.Infof("Dashboard redirector listening on port [%s]", portStr)
+	err = http.ListenAndServe(portStr, nil)
+	if err != nil {
+		logrus.Fatal("ListenAndServe: ", err)
+	}
+}
+
+func redirect(w http.ResponseWriter, r *http.Request) {
+	var errMsg string
+	logrus.Info("Checking for form")
+
+	id := r.FormValue("id")
+	if id == "" {
+		errMsg := "Did not find service instance id as a query param!"
+		logrus.Error(errMsg)
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	logrus.Infof("Got request for service instance %s, looking up dashboard_url", id)
+
+	si, err := crdDao.GetServiceInstance(id)
+	if err != nil {
```

Comment:
What happens if the service instance doesn't exist? Would that result in `err` having a value here? Or would `si` just have the default zero value?

If it's the former, I think the `StatusInternalServerError` below is incorrect in that case and should be a 404, since the instance would have literally been "not found".

If it's the latter, then the log message "Successfully loaded" would be misleading.

Reply:
Updated to check if it's actually an is not found error and will return the correct code in that case, otherwise will return an internal error.

Reply:
:+1: I'd change the value for `errMsg` too just to make it clear that all the software worked correctly, and your instance just doesn't exist. But that's a minor suggestion.

Reply:
updated